portable/NetworkInterface/LPC54018/NetworkInterface.c

```diff
@@ -129,197 +131,222 @@ void ENET_IntCallback( ENET_Type * base,
 
-    /* Active TX/RX. */
-    ENET_StartRxTx( ENET, 1, 1 );
+                if( !link )
+                {
+                    PRINTF( "No Link\n" );
+                    break;
+                }
+            }
 
-    if( xTaskCreate( rx_task, "rx_task", 512, NULL, ( configMAX_PRIORITIES - 1 ), &receiveTaskHandle ) != pdPASS )
-    {
-        PRINTF( "Network Receive Task creation failed!.\n" );
+        /* fall through */
+        case configurePhy:
+           {
+               networkInitialisePhase = configurePhy;
+               enet_config_t config;
+               phy_speed_t speed;
+               phy_duplex_t duplex;
+               PHY_GetLinkSpeedDuplex( &phyHandle, &speed, &duplex );
+               /* Get default configuration 100M RMII. */
+               ENET_GetDefaultConfig( &config );
+
+               /* Use the actual speed and duplex when phy success to finish the autonegotiation. */
+               config.miiSpeed = ( enet_mii_speed_t ) speed;
+               config.miiDuplex = ( enet_mii_duplex_t ) duplex;
+
+               /* Initialize ENET. */
+               uint32_t refClock = 50000000; /* 50MHZ for rmii reference clock. */
+               ENET_Init( ENET, &config, g_macAddr, refClock );
+
+               /* Enable the rx interrupt. */
+               ENET_EnableInterrupts( ENET, ( kENET_DmaRx ) );
+
+               /* Initialize Descriptor. */
+               int bufferIndex;
+
+               for( bufferIndex = 0; bufferIndex < ENET_RXBD_NUM; bufferIndex++ )
+               {
+                   rxbuffer[ bufferIndex ] = ( uint32_t ) &receiveBuffer[ bufferIndex ];
+               }
+
+               /* prepare the buffer configuration. */
+               enet_buffer_config_t buffConfig[ 1 ] =
+               {
+                   {
+                       ENET_RXBD_NUM, ENET_TXBD_NUM,
+                       &g_txBuffDescrip[ 0 ], &g_txBuffDescrip[ 0 ],
+                       &g_rxBuffDescrip[ 0 ], &g_rxBuffDescrip[ ENET_RXBD_NUM ],
+                       &rxbuffer[ 0 ], ENET_BuffSizeAlign( ENET_RXBUFF_SIZE ),
+                   }
+               };
+               ENET_DescriptorInit( ENET, &config, &buffConfig[ 0 ] );
+
+               /* Create the handler. */
+               ENET_CreateHandler( ENET, &g_handle, &config, &buffConfig[ 0 ], ENET_IntCallback, NULL );
+               NVIC_SetPriority( 65 - 16, 4 ); /* TODO this is a hack and I would expect a nice ENET API for priority. */
+
+               /* Active TX/RX. */
+               ENET_StartRxTx( ENET, 1, 1 );
+           }
+
+        case startReceiver:
+            networkInitialisePhase = startReceiver;
+
+            if( xTaskCreate( rx_task, "rx_task", 512, NULL, ( configMAX_PRIORITIES - 1 ), &receiveTaskHandle ) != pdPASS )
+            {
+                PRINTF( "Network Receive Task creation failed!.\n" );
+                break;
+            }
 
-        while( 1 )
-        {
-        }
+            returnValue = pdPASS;
```

Comment:
This is the last statement of the case `startReceiver`. Some compiler will expect a break here:
~~~c
     case startReceiver:
         networkInitialisePhase = startReceiver;
         if( xTaskCreate( ) )
         {
             PRINTF( "Network Receive Task creation failed!.\n" );
             break;
         }
         returnValue = pdPASS;
+        break;
 }
 
 return returnValue;
~~~
Otherwise it all looks good to me.
Thanks, Hein

Reply:
Done